packages/aws-cdk-lib/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -570,7 +589,7 @@ export class InterfaceVpcEndpointAwsService implements IInterfaceVpcEndpointServ
       },
     });
 
-    this.name = `${prefix || defaultEndpointPrefix}.${region}.${name}${defaultEndpointSuffix}`;
+    this.name = `${prefix || defaultEndpointPrefix}.${regionPrefix}${name}${defaultEndpointSuffix}`;
```

Comment:
Nit: adding `+ '.'` in line 578 makes this line look a little weird imo. There is a `.` in front of `${regionPrefix}` but the suffix `.` is added above. Either both `.`'s should be added when you make `regionPrefix` or both should be added here (i prefer here). Unless theres another reason you moved the suffix `.`?

Reply:
The issue is that if the endpoint is global, there will be no region. If I'd kept the region variable, we would have had two `.` between the prefix and the service name. That's why I renamed it to `regionPrefix` and gave it the `.` if it wasn't global. I can revert back to having a `region` variable and only append the `.` if it's not empty/not global, but I would be testing the same thing twice then. Another possibility would be to switch to a `[].filter.join('.')` pattern, it would be more readable imo but a bit excessive given the single modifying condition